AWS/aws-cloudtrail/ingest/parser.yml

```diff
@@ -0,0 +1,111 @@
+  - name: set_ecs_fields
+  - name: set_action_fields
+  - name: set_aws_fields
+
+stages:
+  set_ecs_fields:
+    actions:
+      - set:
+          '@timestamp': '{{parsed_date.datetime}}'
+          event.kind: 'event'
+          event.category: ['network']
+          event.type: ['access']
+          event.dataset: 'cloudtrail'
+          event.action: '{{json_event.message.eventName or json_event.message.insightDetails.eventName}}'
+          event.code: '{{json_event.message.errorCode or json_event.message.insightDetails.errorCode}}'
+          event.reason: '{{json_event.message.errorMessage}}'
+          event.provider: '{{json_event.message.eventSource or json_event.message.insightDetails.eventSource}}'
+          cloud.provider: 'aws'
+          cloud.service.name: 'cloudtrail'
+          cloud.region: '{{json_event.message.awsRegion}}'
+          cloud.account.id: '{{json_event.message.userIdentity.accountId}}'
+          cloud.instance.id: '{{json_event.message.requestParameters.instanceId}}'
+          action.type: '{{json_event.message.eventType}}'
+          action.name: '{{json_event.message.eventName}}'
+          action.outcome: 'success'
+          action.target: 'network-traffic'
+          user_agent.original: '{{json_event.message.userAgent}}'
+          user.id: '{{json_event.message.userIdentity.accountId}}'
+      - set:
+          source.address: '{{json_event.message.sourceIPAddress}}'
+          source.ip: '{{parsed_source.source.ip}}'
+          source.domain: '{{parsed_source.source.domain}}'
+        filter: '{{json_event.message.sourceIPAddress != "AWS Internal"}}'
+      - set:
+          user.name: '{{json_event.message.responseElements.user.userId}}'
+        filter: '{{json_event.message.get("responseElements") != None}}'
+      - set:
+          tls.cipher: '{{json_event.message.tlsDetails.cipherSuite}}'
+          tls.version: '{{json_event.message.tlsDetails.tlsVersion}}'
+        filter: '{{json_event.message.get("tlsDetails") != None}}'
+      - set:
+          event.category: ['iam']
+          event.type: ['change']
+        filter: '{{json_event.message.eventName in ("PasswordUpdated", "EnableMFADevice")}}'
+      - set:
+          event.category: ['authentication']
+        filter: '{{json_event.message.eventName == "ConsoleLogin"}}'
+      - translate:
+        dictionary:
+          Success: ['allowed']
+          Failure: ['denied']
+        mapping:
+          json_event.message.responseElements.ConsoleLogin: event.type
+        filter: '{{json_event.message.eventName == "ConsoleLogin"}}'
+
+  set_action_fields:
+    actions:
+      - set:
+          action.properties.errorCode: '{{json_event.message.errorCode}}'
+          action.properties.errorMessage: '{{json_event.message.errorMessage}}'
+          action.properties.recipientAccountId: '{{json_event.message.recipientAccountId}}'
+          action.properties.userIdentity: '{{json_event.message.userIdentity}}'
```

Comment:
duplicated with `user.id`

Reply:
`action.properties` is available for retro-compatibility purpose